### PR TITLE
Feat/1945 Buttons should work in production

### LINKED ISSUE #1945: As tech lead, I need all frontend app buttons in prod env to point to correct endpoints
**Description:**
As of July 2022, the frontend and backend in the TDP prod environment are located in different domains. the frontend is in an acf domain, while the backend is in a cloudgov domain. The dev and staging env have not been configured to use multiple domains, so we need to ensure that all frontend buttons in prod hit the correct endpoints. 

currently when : 
```
BASE_URL: https://tdp-backend-prod.app.cloud.gov/v1
FRONTEND_BASE_URL: https://tanfdata.acf.hhs.gov
```
- the `request access` button at [tanfdata.acf.hhs.gov/home](tanfdata.acf.hhs.gov/home) does not work. 
- the `submit data files` button at [tanfdata.acf.hhs.gov/data-files](tanfdata.acf.hhs.gov/data-files) does not work. and returns a `403 error`
- the `sign out` button for users with an acf email address redirects to and acf-ams page that returns a  `Invalid redirect uri` and does not successfully logout on first attempt. (_this button functions properly for non-acf email addresses._)

**Acceptance Criteria:** 
_Create a list of functional outcomes that must be achieved to complete this issue_

- [ ] request access button at tanfdata.acf.hhs.gov/home functions properly. 
- [ ] users with appropriate permissions can submit data files via submit data file button
- [ ] all acf and non-acf users can sign out properly from frontend
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

**Tasks:**
_Create a list of granular, specific work items that must be completed to deliver the desired outcomes of this issue_ 

- [ ] reproduce issues in description
- [ ] fix buttons
- [ ] Run Testing Checklist and confirm all tests pass

**Notes:**
- [swagger](https://tdp-backend-prod.app.cloud.gov/swagger/) was used to submit data files to ensure that download button on frontend was functioning properly. [common.py](https://github.com/ADPennington/TANF-app/commits/master) needed changes to facilitate this. 
- frontend sign out button

### PR DESCRIPTION
**Pull request by riatzukiza**
## Summary of Changes

There was a lot of garbage we didn't need in the other pull request, we wanted to start a pr with a fresh history to consolidate.

Pull request closes #1945

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
cd tdrs-backend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d 
```

1. Open http://localhost:3000/ and sign in.
1. Proceed with functional tests as described herein.
1. Test steps should be captured in the demo GIF(s) and/or screenshots below.
> *Demo GIF(s) and screenshots for testing procedure*

## Deliverables
_More details on how deliverables herein are assessed included [here](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [ ] [**_insert ACs here_**]
+ [ ] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [ ] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [ ] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [ ] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [ ] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [ ] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [ ] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

File: .gitconfig [hunk 1]
```diff
@@ -12,5 +12,4 @@
 	allowed = scripts/deploy-backend.sh:.+:DJANGO_SECRET_KEY=..python -c .from secrets import token_urlsafe. print.token_urlsafe..*
 	allowed = .git/config:.*
 	allowed = .gitconfig:.*
```
Comment (andrew-jameson) [inline .gitconfig:14]: Not sure if you meant to delete this.

```suggestion
	allowed = .gitconfig:.*
	allowed = ./tdrs-frontend/node_modules/.*
```

File: scripts/deploy-frontend.sh [hunk 2]
```diff
@@ -41,6 +41,7 @@ update_frontend()
     fi
 
     cf map-route "$CGHOSTNAME_FRONTEND" app.cloud.gov --hostname "${CGHOSTNAME_FRONTEND}"
+    cf map-route "$CGHOSTNAME_FRONTEND" tanfdata.acf.hhs.gov
```
Comment (andrew-jameson) [inline scripts/deploy-frontend.sh:44]: For clarification, wouldn't we only want to run this after we check the `CF_SPACE` first? Or is this change only for testing at this stage?

**Comment by ADPennington** [comment]
@riatzukiza including some videos of the login.gov route for new requestors and admin. great to see our acf backend! 🚀 

cc: @n0remac see the backend owasp report in prod. it references the staging site but i think we want the prod urls to be crawled. 

_new request_

https://user-images.githubusercontent.com/63075587/181853678-5a89d2ba-2130-4500-80d6-59d0ef283005.mov


_sys admin navigation and data submission_ 

https://user-images.githubusercontent.com/63075587/181853765-5a68aa4a-e6c1-4ba8-bf08-a64e66626693.mov

**Comment by n0remac** [comment]
> 

Is that an owasp scan on the uploaded file?

**Comment by andrew-jameson** [comment]
Superseded by #2021

**Pull request closed without merging**